userland/src/parameters.cpp

```diff
@@ -0,0 +1,76 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template<typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
+  } while (status != std::future_status::ready && rclcpp::ok());
+
+  if (std::future_status::ready == status) {
+    return future.get();
+  }
+
+  throw std::runtime_error("Invalid future status");
+}
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+
+  const std::string node_name("parameters_example");
+
+  auto node = rclcpp::Node::make_shared(node_name);
+
+  auto f1 = node->async_has_parameter(node_name, rclcpp::parameter::ParameterQuery("foo"));
+  bool found = wait_for_future<bool>(node, f1);
+  std::cout << "Parameter (foo) found: " << found << std::endl;
```

Comment:
Does this print out `0` or `1`? Or does it do something better for readability? Maybe it needs `(found ? "true" : "false")`?